### PR TITLE
drivers: ad7606: add name field to chip info table (for debugging)

### PR DESCRIPTION
**Pull request by commodo**
## Pull Request Description

Please replace this with a detailed description and motivation of the changes. 
You can tick the checkboxes below with an 'x' between square brackets or just check them after publishing the PR. 
If this PR contains a breaking change, list dependent PRs and try to push all related PRs at the same time.

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [ ] I have commented my code, at least hard-to-understand parts
- [x] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

File: drivers/adc/ad7606/ad7606.c [hunk 1]
```diff
@@ -1122,9 +1133,13 @@ int32_t ad7606_init(struct ad7606_dev **device,
...
 	dev->device_id = init_param->device_id;
-	dev->num_channels = ad7606_chip_info_tbl[dev->device_id].num_channels;
-	dev->max_dout_lines = ad7606_chip_info_tbl[dev->device_id].max_dout_lines;
-	if (ad7606_chip_info_tbl[dev->device_id].has_registers)
+	info = &ad7606_chip_info_tbl[dev->device_id];
+	printf("Initializing device %s, num-channels %u SDI lines %u\n",
+	       info->name, info->num_channels, 1 << info->max_dout_lines);
+
+	dev->num_channels = info->num_channels;
+	dev->max_dout_lines = info->max_dout_lines;
```
Comment (amiclaus) [inline drivers/adc/ad7606/ad7606.c:1141]: maybe it is worth passing directly the chip info structure to the device descriptor (as it is a common approach for linux drivers too). but i guess this is subject for another patch.

**Review by amiclaus — APPROVED** [review]

**Review by amiclaus — COMMENTED** [review]
nit: 14f9b4dfa20a2a0d8426605428895693d0ac3c3e needs a bit of shrinking for commit title

File: drivers/adc/ad7606/ad7606.c (hunk 1 above)
Comment (commodo) [reply drivers/adc/ad7606/ad7606.c:1141]: yeah, that would have been a good idea from the start;

i may add it later.

**Comment by commodo** [comment]
> nit: [14f9b4d](https://github.com/analogdevicesinc/no-OS/commit/14f9b4dfa20a2a0d8426605428895693d0ac3c3e) needs a bit of shrinking for commit title

well, that does fit in the kernel's 75 character limit;
Github may have a shorter character limit for the title;

**Review by amiclaus — APPROVED** [review]

**Pull request merged**